x/epochs/depinject.go

```diff
@@ -0,0 +1,74 @@
+package epochs
+
+import (
+	"fmt"
+	"sort"
+
+	"golang.org/x/exp/maps"
+
+	modulev1 "cosmossdk.io/api/cosmos/epochs/module/v1"
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/depinject"
+	"cosmossdk.io/depinject/appconfig"
+	"cosmossdk.io/x/epochs/keeper"
+	"cosmossdk.io/x/epochs/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+var _ depinject.OnePerModuleType = AppModule{}
+
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (am AppModule) IsOnePerModuleType() {}
+
+func init() {
+	appconfig.RegisterModule(&modulev1.Module{},
+		appconfig.Provide(ProvideModule),
+		appconfig.Invoke(InvokeSetHooks),
+	)
+}
+
+type ModuleInputs struct {
+	depinject.In
+
+	Config      *modulev1.Module
+	Cdc         codec.Codec
+	Environment appmodule.Environment
+}
+
+type ModuleOutputs struct {
+	depinject.Out
+
+	EpochKeeper keeper.Keeper
+	Module      appmodule.AppModule
+}
+
+func ProvideModule(in ModuleInputs) ModuleOutputs {
+	k := keeper.NewKeeper(in.Environment, in.Cdc)
+	m := NewAppModule(in.Cdc, k)
+	return ModuleOutputs{EpochKeeper: k, Module: m}
+}
+
+func InvokeSetHooks(keeper keeper.Keeper, hooks map[string]types.EpochHooksWrapper) error {
```

Comment:
As discussed on our standup review, we should add documentation on this https://github.com/cosmos/cosmos-sdk/issues/19897

Nothing to do here, just noting it.